### PR TITLE
Bug 1252959: Add tooltip to lock icon in Task Inspector

### PR DESCRIPTION
**Pull request by akshitac8**
Added Tooltip for private and public artifacts.
Referred :[Bug 1252959](https://bugzilla.mozilla.org/show_bug.cgi?id=1252959)

### CODE REVIEW

File: src/views/UnifiedInspector/SearchForm.jsx [hunk 1]
```diff
@@ -138,6 +139,13 @@ export default class SearchForm extends PureComponent {
 
         <Col lg={2} md={2} sm={4} style={{ padding: 0 }}>
           <ButtonGroup justified style={{ marginBottom: 15 }}>
+            <Button
```
Comment (helfi92) [inline src/views/UnifiedInspector/SearchForm.jsx]: Is this part of this bug as well?

Comment (akshitac8) [reply src/views/UnifiedInspector/SearchForm.jsx]: No @helfi92. These are the previous commits.The recent updated file is src/components/ArtifactList/index.jsx.  If it is confusing shall I make new pull request?
The referred bug here is [Bug 1252959](https://bugzilla.mozilla.org/show_bug.cgi?id=1252959)

Comment (helfi92) [reply src/views/UnifiedInspector/SearchForm.jsx]: I think rebasing on master should fix the issue.

Comment (helfi92) [reply src/views/UnifiedInspector/SearchForm.jsx]: http://blessing.io/git/git-rebase/open-source/2015/08/23/git-rebasing-to-resolve-conflicts.html

Comment (akshitac8) [reply src/views/UnifiedInspector/SearchForm.jsx]: Ohkay @helfi92 , I think its done.

Comment (eliperelman) [reply src/views/UnifiedInspector/SearchForm.jsx]: I still see this change here, that's not intentional, correct?

File: src/components/ArtifactList/index.jsx [hunk 2]
```diff
@@ -141,16 +141,36 @@ export default class ArtifactList extends PureComponent {
             target="_blank"
             rel="noopener noreferrer"
             key={`runs-menu-artifacts-${index}`}>
-            <i className={`fa fa-${icon}`} style={{ marginRight: 5 }} /> {name}
+            <i
+              className={`fa fa-${icon}`}
+              style={{ marginRight: 5 }}
+              title="public made"
```
Comment (eliperelman) [inline src/components/ArtifactList/index.jsx]: Can you change this to say "This artifact is public."?

File: src/components/ArtifactList/index.jsx [hunk 3]
```diff
@@ -141,16 +141,36 @@ export default class ArtifactList extends PureComponent {
...
     ) : (
       <div style={{ fontSize: 14, ...style }}>
         {artifacts.map(({ name, locked, icon, url }, index) => (
           <div key={`runs-menu-artifacts-${index}`} style={{ marginBottom: 8 }}>
-            {locked && <i className="fa fa-lock" style={{ marginRight: 5 }} />}
-            <i className={`fa fa-${icon}`} style={{ marginRight: 5 }} />
+            {locked && (
+              <i
+                className="fa fa-lock"
+                title="This artifact is private and requires privileges to download it."
```
Comment (eliperelman) [inline src/components/ArtifactList/index.jsx]: Remove the word "it" from the end of the sentence.

File: src/views/UnifiedInspector/SearchForm.jsx (hunk 1 above)
Comment (akshitac8) [reply src/views/UnifiedInspector/SearchForm.jsx]: @eliperelman No thats not intentional. This was due to PR #482 but, I have reverted the unwanted commit for this PR.

**Comment by eliperelman** [comment]
We also have a Tooltip component in use throughout this app, could you change your implementation to use that instead of just titles?

**Comment by akshitac8** [comment]
Ohkay @eliperelman , I will make the changes.

File: src/components/ArtifactList/index.jsx [hunk 4]
```diff
@@ -120,6 +126,14 @@ export default class ArtifactList extends PureComponent {
   render() {
     const { menu, style } = this.props;
     const { artifacts, error } = this.state;
+    const pubtooltip = (
```
Comment (helfi92) [inline src/components/ArtifactList/index.jsx]: Let's rename this to `publicTooltip` to make it more clear.

File: src/components/ArtifactList/index.jsx [hunk 5]
```diff
@@ -120,6 +126,14 @@ export default class ArtifactList extends PureComponent {
...
+      <Tooltip id="public-tooltip">This artifact is public.</Tooltip>
+    );
+    const pritooltip = (
```
Comment (helfi92) [inline src/components/ArtifactList/index.jsx]: `privateTooltip` instead of `pritooltip`.

File: src/views/UnifiedInspector/SearchForm.jsx (hunk 1 above)
Comment (helfi92) [reply src/views/UnifiedInspector/SearchForm.jsx]: I still see it (https://github.com/taskcluster/taskcluster-tools/pull/486/files). Have you tried rebasing on the upstream master branch?

1. Make sure the `upstream` remote is not pointing to your forked repo
```
$ git remote -v
origin	git@github.com:helfi92/taskcluster-tools.git (fetch)
origin	git@github.com:helfi92/taskcluster-tools.git (push)
upstream	https://github.com/taskcluster/taskcluster-tools.git (fetch)
upstream	https://github.com/taskcluster/taskcluster-tools.git (push)
```

As you can see the upstream remote is pointing to `https://github.com/taskcluster/taskcluster-tools.git` which is not where my forked repository is located.

2. Run `git pull upstream master --rebase` to perform a rebase

Let me know if this is still confusing.

Comment (akshitac8) [reply src/views/UnifiedInspector/SearchForm.jsx]: @helfi92 Thank you for the help. I hope it is done correctly?

**Comment by akshitac8** [comment]
Thank you :)

**Pull request merged**